### ISSUE TITLE
Detached Modals are added back to DOM after destroy

### ISSUE THREAD

**Issue by ghost** [jschr/bootstrap-modal#143]
I tend to detach certain sorts of modals when not in use so I don't end up with possible ID conflicts, that are hard to plan for, since my different ajax views can use common templates.

So I usually use the hidden or destroy events to $element.detach() and cache it somewhere.

However, the teardown method will add the element back o the dom if missing.

```
if (this.$parent !== this.$element.parent())
    this.$element.appendTo(this.$parent);
```

My question is why does this exist? I have commented it out and haven't seen any adverse effect...yet.
